cas-server-core-configuration/src/main/java/org/apereo/cas/configuration/model/support/x509/X509Properties.java

```diff
@@ -0,0 +1,187 @@
+package org.apereo.cas.configuration.model.support.x509;
+
+/**
+ * This is {@link X509Properties}.
+ *
+ * @author Misagh Moayyed
+ * @since 5.0.0
+ */
+
+public class X509Properties {
+
+    /**
+     * Default setting to limit the number of intermediate certificates.
+     */
+    private static final int DEFAULT_MAXPATHLENGTH = 1;
+
+    /**
+     * Default setting whether to allow unspecified number of intermediate certificates.
+     */
+    private static final boolean DEFAULT_MAXPATHLENGTH_ALLOW_UNSPECIFIED = false;
+
+    /**
+     * Default setting to check keyUsage extension.
+     */
+    private static final boolean DEFAULT_CHECK_KEYUSAGE = false;
+
+    /**
+     * Default setting to force require "KeyUsage" extension.
+     */
+    private static final boolean DEFAULT_REQUIRE_KEYUSAGE = false;
+
+    private String serialNumberPrefix = "SERIALNUMBER=";
+    private String valueDelimiter = ", ";
+    private int revocationPolicyThreshold = 172800;
+    private boolean checkAll;
+    private int refreshIntervalSeconds = 3600;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['3600' is a magic number.](https://www.codacy.com/app/mmoayyed/cas/file/2749419921/issues/source?bid=3256115&fileBranchId=3357337#l36)